app/uk/gov/hmrc/servicedependencies/service/DeadLetterHandler.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.servicedependencies.service
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.alpakka.sqs.MessageAction.Delete
+import akka.stream.alpakka.sqs.SqsSourceSettings
+import akka.stream.alpakka.sqs.scaladsl.{SqsAckSink, SqsSource}
+import com.github.matsluni.akkahttpspi.AkkaHttpClient
+import com.google.inject.{Inject, Singleton}
+import play.api.Logger
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.Message
+import uk.gov.hmrc.servicedependencies.config.ArtefactReceivingConfig
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
```

Comment:
Not needed - you've got the singleton binding configured in the module using asEagerSingleton.  At best it's not needed here, not sure what the rules are about which one overrides the other.